docs/documentation/head/server-prepare.md

```diff
@@ -9,37 +9,143 @@ nexttitle: Physical and Logical replication API
 next: replication.html
 ---
 
+### Motivation
+
 The PostgreSQL™ server allows clients to compile sql statements that are expected
 to be reused to avoid the overhead of parsing and planning the statement for every
 execution. This functionality is available at the SQL level via PREPARE and EXECUTE
 beginning with server version 7.3, and at the protocol level beginning with server
 version 7.4, but as Java developers we really just want to use the standard
 `PreparedStatement` interface.
 
-### Note
+> Server side prepared statements are planned only once by the server. This avoids
+the cost of replanning the query every time, but also means that the planner
+cannot take advantage of the particular parameter values used in a particular
```

Comment:
there are actually a number of plans that are created, not just one. I would have to find the details, but there is some concept of different plans for different values

Reply:
The implementation is https://github.com/postgres/postgres/blob/9d4649ca49416111aee2c84b7e4441a0b7aa2fac/src/backend/utils/cache/plancache.c#L1034-L1057, however I was unsure if I should modify (update) this comment.

Reply:
Here are the change notes from 9.2:

```Prepared statements used to be optimized once, without any knowledge of the parameters' values. With 9.2, the planner will use specific plans regarding to the parameters sent (the query will be planned at execution), except if the query is executed several times and the planner decides that the generic plan is not too much more expensive than the specific plans.```

Which conflicts with the comment above.

Reply:
Ok, this summary sounds good.